### PR TITLE
Deprecate SendChannel.offer and replace its usages along the code base

### PR DESCRIPTION
**Pull request by qwwdfsad**

### CODE REVIEW

File: ui/kotlinx-coroutines-javafx/src/JavaFxConvert.kt [hunk 1]
```diff
@@ -27,11 +26,11 @@ import kotlinx.coroutines.flow.*
...
-        try {
-            offer(newValue)
-        } catch (e: CancellationException) {
-            // In case the event fires after the channel is closed
-        }
+        /*
+         * Do not propagate an exception to the caller, it
+         * already should've been handle by the downstream
+         */
+        trySend(newValue)
```
Comment (qwwdfsad) [inline ui/kotlinx-coroutines-javafx/src/JavaFxConvert.kt:33]: Could you please take a look @dkhalanskyjb?

File: ui/kotlinx-coroutines-javafx/src/JavaFxConvert.kt [hunk 2]
```diff
@@ -27,11 +26,11 @@ import kotlinx.coroutines.flow.*
 @ExperimentalCoroutinesApi // Since 1.3.x
 public fun <T> ObservableValue<T>.asFlow(): Flow<T> = callbackFlow<T> {
     val listener = ChangeListener<T> { _, _, newValue ->
-        try {
-            offer(newValue)
-        } catch (e: CancellationException) {
-            // In case the event fires after the channel is closed
-        }
+        /*
+         * Do not propagate an exception to the caller, it
```
Comment (dkhalanskyjb) [inline ui/kotlinx-coroutines-javafx/src/JavaFxConvert.kt:30]:
```suggestion
         * Do not propagate the exception to the ObservableValue, it
```

File: ui/kotlinx-coroutines-javafx/src/JavaFxConvert.kt [hunk 3]
```diff
@@ -27,11 +26,11 @@ import kotlinx.coroutines.flow.*
...
+         * already should've been handle by the downstream
```
Comment (dkhalanskyjb) [inline ui/kotlinx-coroutines-javafx/src/JavaFxConvert.kt:31]:
```suggestion
         * already should've been handled by the downstream
```

File: ui/kotlinx-coroutines-javafx/src/JavaFxConvert.kt (hunk 1 above)
Comment (dkhalanskyjb) [reply ui/kotlinx-coroutines-javafx/src/JavaFxConvert.kt:33]: Makes sense.

**Review by elizarov — CHANGES_REQUESTED** [review]
👍 Well done

**Review by elizarov — APPROVED** [review]

**Pull request merged**